### DISCUSSION TITLE
Super High GPU Utilization on Windows using AG Server

### DISCUSSION THREAD

**Discussion by delioroman** [apohl79/audiogridder#1220] [Help]
Hi all,

I got AudioGridder (server) set up on Windows 11 Pro and when loading plugins (such as but not limited to, FabFilter Q3), the GPU utilization shoots up to the 90% +'s. At first, I thought this had to do with the plugin capture streaming and didn't think it was a big deal. Although, the fans and temps on the machines would shoot up as a result of the GPU utilization. Adjusting the screen capture quality from medium to low, from medium to high, using legacy mode, or even down right disabling it entirely (and controlling the plugin via VNC), didn't change anything. Changing the plugin sandbox settings also didn't change anything. Closing the plugin window on the DAW side, also doesn't change the GPU utilization. It just stays pegged at pretty much 100%. Running the same plugins under a DAW or standalone, run perfectly fine on the same machine with basically no GPU utilization. Leading me to believe this has something to do with AG.

As I added more plugins, the GUI on the DAW (plugin) side would become unusable. Resulting in stutters, lags, etc. When I would look on the server side, the entire GUI reacts the same way. Stutters, lags, etc. The most plugins I could get loaded were around 4-5. Then I would start to notice huge lags in the GUI.

So I got curious.

I also have a Mac computer and tested on it. I was curious to see if the same GPU utilization would occur on macOS. It actually performed as expected. No crazy high GPU utilization or spikes. I could load plugins and no lags, stutters etc. Same AG stock settings. Everything scaled as expected. If I added plugin after plugin, CPU utilization would scale accordingly. Indicating that everything is working as it should. I could keep stacking plugins, and everything performed well. If i kept the plugin GUI open on the DAW side, the CPU utilization would go up a little, but upon closing it, it just dropped. Totally expected behavior and I was really happy with this.

Strangely, on Windows, the latest AG is unusable for me. I can basically only load 2 plugins per machine until the entire GUI of the plugin and OS just lags.

When closing AudioGridder on the Windows side or unloading all the plugins on the AG plugin version, GPU utilization just drops back to 0%. Pretty much telling me there's some sort of bug here.

Driver install is fresh and the latest from NVIDIA. Clean install of Windows 11 Pro with basically just AG server running as well as some plugins to test (FabFilter, PA, etc) and Reaper (just a lightweight DAW) just to test the plugin GPU utilization (which ran all plugins 100% fine).

My machine specs:
CPU: Intel 12900
GPU: NVIDIA RTX T1000 8GB
64GB of RAM

Any insight/help would be much appreciated. 

Thanks for all your amazing work!

Here is a screenshot of just 1 instance of FabFilter Pro-R, 1 Pro-Q3, 1 Pro-MB, and 1 Pro-L, running. I could also change to an instance of Samplacity Berlin Strings, or a PA plugin, etc. it's all the same difference. Maybe something to do with 3D rendering and AG server on Windows. Not sure.

BUT: if I load a plugin that doesn't utilize some sort of 3D rendering on the GUI, it runs 100% fine. 0% GPU utilization. One example was LiquidSonics Lustrous Plates. This plugin under AG server ran as expected just like on macOS.

![Screenshot of Task Manager](https://github.com/apohl79/audiogridder/assets/133480702/2a44f909-2fdf-4dda-a710-8e907fa0619a)

**Comment by apohl79** [comment]
Created an investigation ticket for this. Thx a lot for the detailed report!
